kyleteeter/github-gists-app
# Open Forks Feature

### Goals
The goal of this Merge Request is to add functionality to the "Open Forks" button. When the button is clicked, the app should fetch all the forks for this gist and display the avatar of the users who have created the forks. If you click on a given avatar it should open their particular fork on GitHub in another tab. The button is also set to change the inner text so that when the forks are open the button should read "Close Forks". Once you click on the "Close Forks" button the should be removed from memory and the UI. Also, if you open the forks and there are none then you should see a message saying "No Forks"

**_Default View_**
![Screen Shot 2022-01-17 at 6 16 44 PM](https://user-images.githubusercontent.com/29210839/149850322-4c7c7e18-647b-45a7-bb10-47d712a30a98.png)

**_Open Fork_**
![Screen Shot 2022-01-17 at 6 16 55 PM](https://user-images.githubusercontent.com/29210839/149850319-6e297328-5ac9-473e-b40f-06aaf39e82ff.png)

### Test 
To test this featured search for a GitHub user with gists. From there follow these steps:

1. Click "Open Forks" for one of the Gists.
2. Check to see if the Avatars or "No Forks" message appears.
3. If Avatars appear click on the Avatar to make sure it opens the fork in a new tab.
4. If "No Forks" appears click "Close Fork" and try another Gist. 
5. Finally make sure that the button text changes when the forks are open or close.

### Notes:
Currently, the setup only allows you to open the forks for one Gist at a time. If I would change the functionality to allow you to open forks on multiple gists at a time, I think I would change the forks state in the _SearchResults.jsx_ to be an object where I would append the object when I click "Open Fork" with the key being the Gist Id and the value being the Forks Object. From there when I click "Close Fork" on a given gist I would then target that key-value pair in the state and remove it from the forks object stored in the state. Not sure if this is the best way to go about something like that. 

On the other side of things, I am not 100% sure if I am reloading all of the cards because before I added the ternary operator on line 56 in _SearchResults.jsx_ it was displaying the forks on every single card.
